docs/rules/prefer-in-document.md

```diff
@@ -0,0 +1,27 @@
+# Prefer .toBeInTheDocument in favor of .toHaveLength(1) (prefer-in-document)
+
+## Rule Details
+
+This rule enforces checking existance of DOM nodes using `.toBeInTheDocument()`
+instead of asserting on the length of a query result using `.toHaveLength(1)`.
+
+Examples of **incorrect** code for this rule:
+
+```js
+expect(screen.queryByText("foo")).toHaveLength(1);
+```
+
+Examples of **correct** code for this rule:
+
+```js
+expect(screen.queryByText("foo")).toBeInTheDocument();
+```
```

Comment:
can we add a few more examples?  like `expect(screen.getAllByText("foo")).toHaveLength(2)` for example.

Reply:
same as below, queryBy should be used to check that something is not in the document, get by should used to check that it _is_ in the document. let's update the tests & docs to reflect that please.

Reply:
I've added a few more examples for both valid and invalid cases. 😊